internal/core/software_renderer.rs

```diff
@@ -1000,14 +1001,24 @@ enum SceneCommand {
 }
 
 struct SceneTexture<'a> {
+    /// This should have a size so that the entire slice is ((height - 1) * pixel_stride + width) * bpp
     data: &'a [u8],
     format: PixelFormat,
     /// number of pixels between two lines in the source
     pixel_stride: u16,
-    source_size: PhysicalSize,
+
+    extra: SceneTextureExtra,
+}
+
+#[derive(Clone, Copy, Debug)]
+struct SceneTextureExtra {
+    dx: fixed::Fixed<u16, 8>,
```

Comment:
I suggest either a longer name or a comment for dx/dy/off_x/off_y.

Reply:
I like the short names because that's the name that appears in the math formula usually.
I should have named it `Δx` and `Δy` :-)
But you're entirely right about the documentation. I added that.